skcosmo/feature_selection/_greedy.py

```diff
@@ -0,0 +1,154 @@
+"""
+Sequential feature selection
+"""
+import numbers
+
+import numpy as np
+
+from sklearn.feature_selection._base import SelectorMixin
+from sklearn.base import BaseEstimator, MetaEstimatorMixin
+from sklearn.utils.validation import check_is_fitted, check_array
+
+
+class GreedySelector(SelectorMixin, MetaEstimatorMixin, BaseEstimator):
+    """Transformer that performs Greedy Feature Selection.
+
+    This Greedy Selector adds (forward selection) features to form a
+    feature subset in a greedy fashion. At each stage, the model scores each
+    feature (without an estimator) and chooses the feature with the maximum score.
+
+    Parameters
+    ----------
+
+    n_features_to_select : int or float, default=None
+        The number of features to select. If `None`, half of the features are
+        selected. If integer, the parameter is the absolute number of features
+        to select. If float between 0 and 1, it is the fraction of features to
+        select.
+
+    scoring : str, callable, list/tuple or dict, default=None
+        A single str (see :ref:`scoring_parameter`) or a callable
+        (see :ref:`scoring`) to evaluate the features. It is assumed that the
+        next feature to select is that which maximizes the score.
+
+        NOTE that when using custom scorers, each scorer should return a single
+        value. Metric functions returning a list/array of values can be wrapped
+        into multiple scorers that return one value each.
+
+    Attributes
+    ----------
+    n_features_to_select_ : int
+        The number of features that were selected.
+
+    support_ : ndarray of shape (n_features,), dtype=bool
+        The mask of selected features.
+
+    """
+
+    def __init__(
+        self,
+        scoring,
+        n_features_to_select=None,
+    ):
+
+        self.n_features_to_select = n_features_to_select
+        self.scoring = scoring
```

Comment:
I agree with you, that we should be consistent with the `sklearn`, and if they don't have some "restart", we also should avoid it. But maybe we can to argue a little with the `sklearn` developers?:) How do you think, is it important ti have a restart module? Also @ceriottm position is interesting